src/rdkafka.c

```diff
@@ -1356,7 +1356,16 @@ rd_kafka_t *rd_kafka_new (rd_kafka_type_t type, rd_kafka_conf_t *app_conf,
         }
 #endif
 
-        if (type == RD_KAFKA_CONSUMER) {
+        if (type == RD_KAFKA_PRODUCER) {
+                if (validate_scope && (conf->scope & _RK_CONSUMER)) {
+                    rd_snprintf(errstr, errstr_size,
+                                "Wrong scope: configuration provided is intended for consumers.");
```

Comment:
keep below 80 columns

Reply:
Ok will shorten...although the coding style guide said not to exceed by too much the 80 col mark.